.github/workflows/unit-test.yml

```
@@ -39,17 +39,21 @@ jobs:
               with:
                   node-version: ${{ matrix.node }}
 
+            - name: Get number of CPU cores
+              uses: SimenB/github-actions-cpu-cores@v2
```

```suggestion
              uses: SimenB/github-actions-cpu-cores@97ba232459a8e02ff6121db9362b09661c875ab8 # v2.0.0
```
For third party actions, we should hardcode the SHA value for the version we wish to use. This is recommended in the [security hardening guidelines](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). We use Dependabot to manage these updates automatically, so the burden of maintenance is not that great.